OpenRA.Mods.Common/Activities/RemoveSelf.cs

```diff
@@ -15,11 +15,12 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class RemoveSelf : Activity
 	{
-		public override Activity Tick(Actor self)
+		public override bool Tick(Actor self)
 		{
-			if (IsCanceling) return NextActivity;
+			if (IsCanceling) return true;
 			self.Dispose();
-			return null;
+			Cancel(self);
```

Comment:
What does cancelling achieve here?

Reply:
It removes the rest of the activity queue, so it achieves the same effect as returning `null` would have before.